partd/pandas.py

```diff
@@ -129,7 +141,7 @@ def block_to_header_bytes(block):
     elif is_datetime64tz_dtype(block):
         extension = ('datetime64_tz_type', (block.values.tzinfo,))
         values = values.view('i8')
-    elif is_extension_array_dtype(block.dtype):
+    elif is_extension_array_dtype(block.dtype) or is_extension_array(values):
```

Comment:
I suppose you could also replace this with `elif isinstance(values, ExtensionArray)`, and I would think this should work for older versions of pandas as well.

(but the current PR should work as well, I think)

Reply:
Thanks for reviewing @jorisvandenbossche! I avoided a pure `isinstance(values, ExtensionArray)` check since `partd` supports older versions of `pandas` which predate the `ExtensionArray` class

Reply:
OK, sounds good then!

Reply:
could also do a `not isinstance(values, np.ndarray)` (unless you support pandas so old that it can still be a CategoricalIndex or something)